packages/nuxt/src/app/plugins/view-transitions.client.ts

```diff
@@ -9,9 +9,8 @@ export default defineNuxtPlugin((nuxtApp) => {
 
   const router = useRouter()
 
-  router.beforeResolve((to) => {
-    if (to.meta.pageTransition === false) { return }
-
+  router.beforeResolve((to, from) => {
+    if (to.path === from.path || to.meta.pageTransition === false) { return }
```

Comment:
```suggestion
    if (to.fullPath === from.fullPath || to.meta.pageTransition === false) { return }
```

Reply:
I tried using `fullPath`. But this causes a transition to be triggered when clicking on the anchor when navigating within the page. The page still gets stuck for a few seconds.

Reply:
Yes, that's a fair point. Can we instead use the matching algorithm e.g. 
https://github.com/nuxt/nuxt/blob/9f5130d06b3d449dd16d33ee5e41c820d037e2c4/packages/nuxt/src/app/components/nuxt-loading-indicator.ts#L43-L47

Reply:
I tried it, and it does work.

Reply:
I have recommit the code with reference to indicator's implementation, thanks for the tip!

Reply:
perfect 👌